### PR TITLE
Enhancement: Add support for RSA encryption key imports in JavaKeystoreKeyProvider

### LINKED ISSUE #29852: Support RSA Encryption Key-Imports for the JavaKeystoreKeyProvider
### Description

Keycloak uses these 3 RSA based algorithms for key encryption:

1. RSAES-PKCS1-v1_5(RSA1_5)
2. RSAES OAEP using default parameters (RSA-OAEP)
3. RSAES OAEP 256 using SHA-256 and MFG1 (RSA-OAEP-256)

The JavaKeystoreKeyProvider curently supports loading RSA keys associated with RSA based signature algorithms (PS256, PS384, PS512, RS256, RS384, RS512).

For the purpose of managing key pairs externally and across servers, asymetric encryption keys shall be imported through the JavaKeystoreKeyProvider. 

### Discussion

_No response_

### Motivation

Simplify key management and key rotation, including the sharing and rotation of symmetric keys.

### Details

Identical to the extension of JavaKeystoreKeyProvider with ECDSA import capability.

See: https://github.com/keycloak/keycloak/issues/26936

### PR DESCRIPTION
**Pull request by francis-pouatcha**
Adds functionality to the JavaKeystoreKeyProvider to allow for the import of RSA keys specifically designated for encryption purposes. This enhancement expands the key management capabilities of Keycloak.

Closes: #29852

### CODE REVIEW

**Comment by ahus1** [comment]
@shawkins / cc: @vmuzikar - maybe this PR is of interest for you as it is about key handling.

**Comment by mposolda** [comment]
@rmartinc Can you review this one please whenever you have a chance?

File: services/src/main/java/org/keycloak/keys/JavaKeystoreKeyProvider.java [hunk 1]
```diff
@@ -93,7 +93,8 @@ protected KeyWrapper loadKey(RealmModel realm, ComponentModel model) {
             String keyAlias = model.get(JavaKeystoreKeyProviderFactory.KEY_ALIAS_KEY);
 
             return switch (algorithm) {
-                case Algorithm.PS256, Algorithm.PS384, Algorithm.PS512, Algorithm.RS256, Algorithm.RS384, Algorithm.RS512 ->
+                case Algorithm.PS256, Algorithm.PS384, Algorithm.PS512, Algorithm.RS256, Algorithm.RS384, Algorithm.RS512,
+                        Algorithm.RSA_OAEP, Algorithm.RSA1_5, Algorithm.RSA_OAEP_256 ->
```
Comment (rmartinc) [inline services/src/main/java/org/keycloak/keys/JavaKeystoreKeyProvider.java:97]: I think that you need to add the `RSA_OAEP` to the property `algorithm`  in the factory ([here](https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/keys/JavaKeystoreKeyProviderFactory.java#L120)). Right? (If not I think you cannot configure it via admin console.)

**Review by rmartinc — CHANGES_REQUESTED** [review]
Thanks @francis-pouatcha! If I'm not wrong, you need to add the new algorithm to the supported list of algorithms of the `JavaKeystoreKeyProviderFactory`, in order to configure it using the console.

File: services/src/main/java/org/keycloak/keys/JavaKeystoreKeyProvider.java (hunk 1 above)
Comment (francis-pouatcha) [reply services/src/main/java/org/keycloak/keys/JavaKeystoreKeyProvider.java:97]: Thanks for the hint @rmartinc. Fixed with [3f33dcf](https://github.com/keycloak/keycloak/pull/29853/commits/3f33dcfee1a6c6dc262c0bcf869362af371ca516).

**Review by rmartinc — APPROVED** [review]
Thanks @francis-pouatcha! LGTM!

**Review by mposolda — APPROVED** [review]
@francis-pouatcha @rmartinc Thanks for handling this and review!

**Pull request merged**